utils/sites.js

```diff
@@ -282,37 +254,99 @@ const updateParticipantData = async (req, res, authObj) => {
+                        const occurrenceResult = handleCancerOccurrences(flatDataObj, dataObj, existingOccurrences, requiredOccurrenceRules);
+                        if (occurrenceResult.error) {
+                            error = true;
+                            responseArray.push({'Invalid Request': {'Token': participantToken, 'Errors': occurrenceResult.message}});
+                            continue participantLoop;
+                        }
+                    }
+                    // TODO: extend for userProfile history.
+                } else {
+                    flatDataObj[path] = [...docData[path], ...dataObj[path]];
+                }
+            }
+        }
+
+        
+        // Handle updates to query.firstName and query.lastName arrays (these are used for participant search). firstName, prefName, and lastName. Consent name does not get updated.
+        // Note: Name fields can be updated directly. Query.name fields are managed. Do not allow direct updates to query.name fields. Add the updated query array to flatDataObj.
+        if (dataObj['query']) {
+            error = true;
+                responseArray.push({'Invalid Request': {'Token': participantToken, 'Errors': 'Query variables cannot be directly updated through this API.'}});
+                continue;
         }
 
-        if(updatedData['399159511']) updatedData[`query.firstName`] = dataObj['399159511'].toLowerCase();
-        if(updatedData['996038075']) updatedData[`query.lastName`] = dataObj['996038075'].toLowerCase();
+        if (dataObj[fieldMapping.firstName] || dataObj[fieldMapping.preferredName] || dataObj[fieldMapping.lastName]) {
+            flatDataObj['query'] = { ...docData['query'] };
+            if (dataObj[fieldMapping.firstName] || dataObj[fieldMapping.preferredName]) {
+                let queryFirstNameArray = docData['query']['firstName'] || [];
+                if (dataObj[fieldMapping.firstName]) updateQueryNameArray(dataObj[fieldMapping.firstName], docData[fieldMapping.firstName], queryFirstNameArray);
+                if (dataObj[fieldMapping.preferredName]) updateQueryNameArray(dataObj[fieldMapping.preferredName], docData[fieldMapping.preferredName], queryFirstNameArray);
+                flatDataObj['query']['firstName'] = queryFirstNameArray;
+            }
+            if (dataObj[fieldMapping.lastName]) {
+                let queryLastNameArray = docData['query']['lastName'] || [];
+                updateQueryNameArray(dataObj[fieldMapping.lastName], docData[fieldMapping.lastName], queryLastNameArray);
+                flatDataObj['query']['lastName'] = queryLastNameArray;
+            }
+        }
 
         console.log("UPDATED DATA");
-        console.log(updatedData);
-
-        if(Object.keys(updatedData).length > 0) {
+        console.log(flatDataObj);
 
+        if (Object.keys(dataObj).length > 0) {
```

Comment:
should this be `flatDataObj` or does it matter?

Reply:
Great call - yes. Updated.